### PR TITLE
cross-platform: fix memmove for overlapping memory cases

### PR DESCRIPTION
**Pull request by obastemur**
Also added a simple sort test to trigger hybrid_sort implementation

### CODE REVIEW

File: pal/src/safecrt/memmove_s.c [hunk 1]
```diff
@@ -6,48 +6,92 @@
...
+    void * dst,
+    const void * src,
+    size_t count
+)
+{
+#if defined(__APPLE__) || defined(__FreeBSD__)
+    if (src <= dst && src + count > dst)
+    {
+        char *temp = (char*) malloc(count);
+        _VALIDATE_RETURN_ERRCODE(temp != NULL, NULL);
```
Comment (digitalinfinity) [inline pal/src/safecrt/memmove_s.c]: Did you mean to supply a return code for the second parameter?

**Comment by digitalinfinity** [comment]
:shipit:

File: pal/src/safecrt/memmove_s.c (hunk 1 above)
Comment (obastemur) [inline pal/src/safecrt/memmove_s.c]: I've just reused the definition while it was fitting

**Comment by obastemur** [comment]
@digitalinfinity Thanks for the review

**Pull request merged**